core/components/scripts/scripts.tsx

```diff
@@ -0,0 +1,59 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import Script from 'next/script';
+import { ComponentProps } from 'react';
+
+import { FragmentOf } from '~/client/graphql';
+
+import { ScriptsFragment } from './fragment';
+
+type ScriptsData = FragmentOf<typeof ScriptsFragment>;
+
+interface ScriptRendererProps {
+  scripts: ScriptsData['headerScripts'] | null;
+  strategy: ComponentProps<typeof Script>['strategy'];
+}
+
+export const ScriptManagerScripts = ({ scripts, strategy }: ScriptRendererProps) => {
+  if (!scripts?.edges) return null;
+
+  const scriptNodes = removeEdgesAndNodes(scripts);
+
+  return (
+    <>
+      {scriptNodes.map((script) => {
```

Comment:
🍹 Only if you want to update this... but a better practice, which might seem weird, is to do two `.map`'s here. One to build the props, then another one to render. It allows each `.map` to have separate concerns.
```tsx
{scriptNodes
  .map((script) => {
    // ...
    return {
      src: "...",
      // ....
    };
  })
  .map((script) => {
    return (
      <Script key={script.id} {...script} />
    );
  })
}
```